src/vs-bicep/Bicep.VSLanguageServerClient/DidChangeWatchedFilesNotifier.cs

```diff
@@ -0,0 +1,135 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Threading.Tasks;
+using Microsoft.VisualStudio.ComponentModelHost;
+using Microsoft.VisualStudio.LanguageServer.Protocol;
+using Microsoft.VisualStudio.Shell;
+using Microsoft.VisualStudio.Shell.Interop;
+using Microsoft.VisualStudio.Workspace.VSIntegration.Contracts;
+using StreamJsonRpc;
+
+namespace Bicep.VSLanguageServerClient
+{
+    /// <summary>
+    /// VS lsp doesn't support 'workspace/didChangeWatchedFiles' notification. This class
+    /// watches workspace files with extension - arm, bicep, json, jsonc and sends across
+    /// notification to language server whenever there is a file change/create/delete event.
+    /// </summary>
+    public class DidChangeWatchedFilesNotifier : IDisposable
+    {
+        private readonly string[] filters = { "*.arm", "*.bicep", "*.json", "*.jsonc" };
```

Comment:
Would be better to pass in the filters into the constructor and keep it generic (including the summary comments)

Reply:
We can definitely pass it into constructor, but I don't understand what you mean by generic. These are the only ones watched in language server command handler and in vscode client.ts file. Am I missing something?

Reply:
Included this scenario in .gif attached to description.

Reply:
Fair enough